envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/RoutesAssertions.kt

```diff
@@ -61,11 +61,13 @@ fun RouteConfiguration.hasNoResponseHeaderToAdd(key: String): RouteConfiguration
 
 fun VirtualHost.hasStatusVirtualClusters(): VirtualHost {
     return this.hasVirtualClustersInOrder(
-        { it.headersList.contains(HeaderMatcher.newBuilder()
-                .setName(":path").setPrefixMatch("/status/").build()) && it.name == "status"
+        {
+            it.headersList[0] == HeaderMatcher.newBuilder().setName(":path").setPrefixMatch("/status/").build() &&
```

Comment:
IMO this is no better than `contains`, because still the test will pass if there will be another header. IMO we should test it like this: `it.headersList == listOf(HeaderMatcher.newBuilder().setName(":path").setPrefixMatch("/status/").build())`